poller/poller.go

```diff
@@ -173,44 +173,61 @@ func (poller *Poller) getSerializedDataSubset(splitNames []string) string {
 func (poller *Poller) getUpdatedSerializedDataSubsets(newSplitData SplitData) map[string]string {
 	updatedSubsets := poller.getCachedSerializedDataSubsets()
 	for key := range updatedSubsets {
-		updatedSubsets[key] = generateSerializedData(newSplitData, strings.Split(key, "."))
+		updatedSubsets[key] = poller.generateSerializedData(newSplitData, strings.Split(key, "."))
 	}
 	return updatedSubsets
 }
 
 // generateSerializedData takes SplitData and generates a script tag
 // that saves the SplitData info to the window object of the browser
-func generateSerializedData(splitData SplitData, splitNames []string) string {
+func (poller *Poller) generateSerializedData(splitData SplitData, splitNames []string) string {
 	if reflect.DeepEqual(splitData, SplitData{}) {
 		return emptyCacheLoggingScript
 	}
 	splitNamesToSerializedData := map[string]string{}
+	splitsSubset := map[string]dtos.SplitDTO{}
 
 	// Serialize values for splits
-	for _, split := range splitData.Splits {
-		index := sort.SearchStrings(splitNames, split.Name)
-		splitIsInSplitNames := index < len(splitNames) && splitNames[index] == split.Name
-		// if the split is not in the splitNames array, do not serialize the split
-		if len(splitNames) > 0 && !splitIsInSplitNames {
-			continue
+	for name, split := range splitData.Splits {
+		if len(splitNames) > 0 {
```

Comment:
nit: can we specify a variable for len(splitNames) as since it's used in multiple places